src/reason-parser/reason_pprint_ast.ml

```diff
@@ -5603,9 +5606,6 @@ let printer = object(self:'self)
                 match x'.pexp_desc with
                 | Pexp_let _ ->
                   Some (makeLetSequence (self#letList x))
-                | Pexp_function l when (pipe || semi) ->
```

Comment:
this match was useless. `Pexp_function` would be caught by `expression_immediate_extension_sugar`